### DISCUSSION TITLE
Cannot setup Modbus TCP

### DISCUSSION THREAD

**Discussion by Captain0Kangaroo** [nathanmarlor/foxess_em#335] [General]
Just re-enabled after having been in FLUX for the summer and i needed to update my Solcast API. 

I cant move forward to save it because it's asking for the network connection again but when i hit submit i get this:

<img width="1023" alt="image" src="https://github.com/nathanmarlor/foxess_em/assets/96144685/5b16fe5c-3e8b-46bc-992f-42ee17a0b470">

Any ideas how to move forward ?

Fox Modbus is working fine BTW.

**Comment by nathanmarlor** [comment]
@Captain0Kangaroo was it working previously? Any logs before/after the error?

**Reply by Captain0Kangaroo** [reply]
Yeah it was working fine before I disabled it back in May.
Remind me how to get the logs :-)

**Reply by nathanmarlor** [reply]
Settings -> System -> Logs.

Which version of HA?

**Reply by Captain0Kangaroo** [reply]
Home Assistant Core
ConnectionException()
18:09:54 – (WARNING) FoxESS - Energy Management (custom integration) - message first occurred at 17:09:04 and shows up 6 times
ConnectionException()
17:15:42 – (WARNING) FoxESS - Energy Management (custom integration) - message first occurred at 17:09:31 and shows up 2 times
TypeError("'NoneType' object is not subscriptable")
17:15:28 – (ERROR) FoxESS - Energy Management (custom integration) - message first occurred at 17:07:48 and shows up 4 times

**Reply by Captain0Kangaroo** [reply]
Just changed version to the one below and it's working.

**Reply by nathanmarlor** [reply]
Which version below?

**Reply by Captain0Kangaroo** [reply]
1.7.6

**Reply by nathanmarlor** [reply]
And which version of HA?

**Reply by Captain0Kangaroo** [reply]
Home Assistant 2023.9.0
Supervisor 2023.08.3
Operating System 10.5
Frontend 20230906.1 - latest

**Reply by Captain0Kangaroo** [reply]
I just updated to 1.8.0 and its working but i didn't need to enter in my details.

**Reply by nathanmarlor** [reply]
Very strange, 1.8.0 was released to fix an update to the Pandas library. Can you check which version of Pandas you're running please? 

From the command line:

> python3
> import pandas as pd
> print(pd.__version__)

https://www.educative.io/answers/how-to-check-the-pandas-version-in-python

Or similar, I'm on mobile ATM.

Thanks!

**Reply by Captain0Kangaroo** [reply]
I'm out of my depth there! No idea even after reading that link :-)

**Reply by morton027** [reply]
I get this same error, it's fine when I set up the integration first time, but cannot get into configuration to modify any settings afterwards.

The quick fix I found was to get hold of a laptop (I don't own one, just mobile and tablet) to change the IP address of my Modbus then change back to the one used.

My HA version is as follows:

Home Assistant 2023.9.2
Supervisor 2023.09.2
Operating System 10.5
Frontend 20230911.0 - latest

**Comment by miralemR** [comment]
I have EW11 (TCP server) modbus and I connect to it with rs485 through Foxess-Modbus integration with the IP. In Foxess energy management when I insert the same IP address it gives an error. What am I doing wrong / is there any workaround or...?

**Reply by bmgmac** [reply]
I too am having the same issue using an EW11 adaptor. 
<img width="488" alt="image" src="https://github.com/user-attachments/assets/b628e768-9912-4177-9ee4-98b89a32d15e">

2024-09-24 00:39:10.372 DEBUG (MainThread) [custom_components.foxess_em.fox.fox_modbus] Connecting to modbus - ({'connection_type': 'Modbus TCP', 'host': '192.168.0.88', 'port': '502'})
2024-09-24 00:39:10.372 DEBUG (MainThread) [custom_components.foxess_em.fox.fox_modbus] Reading register: (40002, 1, 247)
2024-09-24 00:39:10.510 DEBUG (MainThread) [custom_components.foxess_em.fox.fox_modbus] Closing connection to modbus
2024-09-24 00:39:10.511 WARNING (MainThread) [custom_components.foxess_em.config_flow] ModbusIOException('[Input/Output] Error reading registers: Exception Response(131, 3, IllegalAddress)')
